src/shogun/mathematics/linalg/SGLinalg.cpp

```diff
@@ -118,8 +116,32 @@ bool SGLinalg::hasGPUBackend() const
 	return m_gpubackend != nullptr;
 }
 
+template <class T>
+std::shared_ptr<BaseVector<T>> SGLinalg::convert_vector(const SGVector<T> &vector) const
+{
+	std::shared_ptr<BaseVector<T>> basevector;
+
+#ifdef HAVE_VIENNACL
+	if hasGPUBackend()
+	{
+		basevector = std::shared_ptr<BaseVector<T>>(new GPUVector<T>(vector));
+	}
+	else
+	{
+		SG_SINFO("User did not register GPU backend. Will use CPU backend instead. \n");
```

Comment:
See my other comments about this error message. Suboptimal to write this everytime
@lambday @vigsterkr